File: app/src/test/java/com/soonyong/mywebcrawler/config/CrawlConfigTest.java
package com.soonyong.mywebcrawler.config;


import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class CrawlConfigTest {

    private static final ObjectMapper objectMapper = new ObjectMapper();

    @Test
    public void testDeserialize() throws JsonProcessingException {
        // given
        String content = "{" +
                "   \"targets\" : " +
                "   [" +
                "       {" +
                "           \"active\":true," +
                "           \"interval\":1000," +
                "           \"url\":\"https://www.google.com\"," +
                "           \"targetXPath\":\"myXPath\"" +
                "       }" +
                "   ]" +
                "}";

        // when
        CrawlConfig crawlConfig = objectMapper.readValue(content, CrawlConfig.class);

        // then
        assertEquals(1, crawlConfig.getTargets().size());
        CrawlConfig.Target target = crawlConfig.getTargets().get(0);
        assertEquals("https", target.getUrl().getScheme());
        assertEquals("www.google.com", target.getUrl().getHost());
        assertTrue(target.isActive());
        assertEquals(1000, target.getInterval());
        assertEquals("myXPath", target.getTargetXPath());
    }
}

File: app/src/test/java/com/soonyong/mywebcrawler/config/CrawlConfigTest.kt
package com.soonyong.mywebcrawler.config

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.Assert
import org.junit.Assert.assertEquals
import org.junit.Test

class CrawlConfigTest {
    @Test
    @Throws(JsonProcessingException::class)
    fun testDeserialize() {
        // given
        val content = "{" +
                "   \"targets\" : " +
                "   [" +
                "       {" +
                "           \"title\":\"my title\"" +
                "           \"active\":true," +
                "           \"interval\":1000," +
                "           \"url\":\"https://www.google.com\"," +
                "           \"targetXPath\":\"myXPath\"" +
                "       }" +
                "   ]" +
                "}"

        // when
        val crawlConfig = objectMapper.readValue(content, CrawlConfig::class.java)

        // then
        assertEquals(1, crawlConfig.targets.size)
        val target: CrawlConfig.Target = crawlConfig.targets[0]
        assertEquals("https", target.url.scheme)
        assertEquals("www.google.com", target.url.host)
        Assert.assertTrue(target.active)
        assertEquals(1000, target.interval)
        assertEquals("myXPath", target.targetXPath)
        assertEquals("my title", target.title)
    }

    companion object {
        private val objectMapper = ObjectMapper()
    }
}